tools/scalajsenv.js

```diff
@@ -761,21 +777,24 @@ ScalaJS.TypeData.prototype.initClass = function(
 initClass(
 //!endif
     internalNameObj, isInterface, fullName,
-    ancestors, isRawJSType, parentData, isInstance, isArrayOf) {
+    ancestors, typeTag, isRawJSType, parentData, isInstance, isArrayOf) {
   const internalName = ScalaJS.propertyName(internalNameObj);
 
   isInstance = isInstance || function(obj) {
-    return !!(obj && obj.$classData && obj.$classData.ancestors[internalName]);
+    let data = obj && ScalaJS.ClassData[obj.$typeTag];
```

Comment:
This is awkward if `$typeTag` does not exist on `obj` (happens for primitives and JS objects). It would be good to use
```js
ScalaJS.isScalaJSObject(obj) ? ScalaJS.ClassData[obj.$typeTag] : null;
```